fix: Issue #554 fix audio input test case in docker container

jimthompson5802 opened the pull request:
# Code Pull Requests

Fix #554 

Tested fix as follows:
```
docker build -t ludwig_test .
docker run -it --rm -v $PWD:/opt/project -w /opt/project --entrypoint /bin/bash ludwig_test
cd tests
pytest -v 
```

All pytest test cases, including the `test_experiment_audio_inputs` case completed successfully.  Log file for `pytest -v`.
[docker_after_fix_sndfile_pytest_log.txt](https://github.com/uber/ludwig/files/3777048/docker_after_fix_sndfile_pytest_log.txt)

jimthompson5802 commented:
@w4nderlust   Removed pytest from the Dockerfile.

jimthompson5802 commented:
@w4nderlust 

I added the purpose specific Dockerfiles as we discussed.  

|File|Description|
|-----------|----------|
|Dockerfile|Basic version (requirements.txt only)|
|Dockerfile_full|Full Version + install libsndfile1|
|Dockerfile_dev|Full Version + install libsndfile1 + test requirements|
|Dockerfile_gpu|Basic Version modified to provide gpu support|

w4nderlust commented:
We updated to TensorFlow 1.15.0 that doesn't require a specific gpu package install, can you please update the pr accordingly?

jimthompson5802 commented:
Per request revised the `Dockerfile_gpu` to reflect TF 1.15.0.

Instead of using the `latest-gpu-py3` tag specified `1.15.0-gpu-py3` tag.  Use of `latest-gpu-py3` led to this error message during docker build
```
#6 80.57 ERROR: tensorflow-gpu 2.0.0 has requirement tensorboard<2.1.0,>=2.0.0, but you'll have tensorboard 1.15.0 which is incompatible.
#6 80.57 ERROR: tensorflow-gpu 2.0.0 has requirement tensorflow-estimator<2.1.0,>=2.0.0, but you'll have tensorflow-estimator 1.15.1 which is incompatible.
#6 80.57 Installing collected packages: Cython, python-dateutil, pytz, pandas, scipy, tabulate, joblib, scikit-learn, tqdm, tensorflow-estimator, tensorboard, tensorflow, PyYAML
#6 86.62   Found existing installation: tensorflow-estimator 2.0.0
#6 86.66     Uninstalling tensorflow-estimator-2.0.0:
#6 86.72       Successfully uninstalled tensorflow-estimator-2.0.0
#6 86.97   Found existing installation: tensorboard 2.0.0
#6 87.06     Uninstalling tensorboard-2.0.0:
#6 87.21       Successfully uninstalled tensorboard-2.0.0
#6 106.0 Successfully installed Cython-0.29.14 PyYAML-5.2 joblib-0.14.1 pandas-0.25.3 python-dateutil-2.8.1 pytz-2019.3 scikit-learn-0.22 scipy-1.4.1 tabulate-0.8.6 tensorboard-1.15.0 tensorflow-1.15.0 tensorflow-estimator-1.15.1 tqdm-4.41.0
```

w4nderlust commented:
Hi @jimthompson5802 , the gpu dockerfile is no longer needed, and i would keep just the regular (with the full requirements minus the test ones) and the test ones (full + test). Does that make sense to you?

jimthompson5802 commented:
@w4nderlust I believe I made the changes you requested:
* Deleted `Dockerfile_gpu`
* Updated the "regular" `Dockerfile` to include the full set of dependent packages
* Left the `Dockerfile_dev` alone.  It contains both the full and test related dependent packages.

Let me know if this is correct.

w4nderlust commented:
thank you!
